openshift/topics/advanced_concepts.adoc

```diff
@@ -282,12 +282,81 @@ route "sso" patched
 
 Tip: The easiest way to get started is to deploy {project_name} from a template and then modify the Endpoints object. You might also need to update some of the datasource configuration variables in the DeploymentConfig. Once you're done, just roll a new deployment out.
 
+[[sso-using-custom-jdbc-driver]]
+=== Using Custom JDBC Driver
+
+To connect to any database, the JDBC driver for that database must be present and {project_name} configured
+properly. Currently, the only JDBC driver available in the image is the PostgreSQL JDBC driver. For any other
+database, you need to extend the {project_name} image with a custom JDBC driver and a CLI script to register
+it and set up the connection properties. The following steps illustrate how to do that, taking MariaDB driver
+as an example. Update the example for other database drivers accordingly.
+
+. Create an empty directory.
+. Download the JDBC driver binaries into this directory.
+. Create a new `Dockerfile` file in this directory with the following contents. For other databases, replace `mariadb-java-client-2.5.4.jar` with the filename of the respective driver:
++
+[source,subs="attributes+,macros+,+quotes"]
+----
+*FROM* {openshift_image}:latest
+
+*COPY* sso-extensions.cli /opt/eap/extensions/
+*COPY* mariadb-java-client-2.5.4.jar /opt/eap/extensions/jdbc-driver.jar
+----
+. Create a new `sso-extensions.cli` file in this directory with the following contents. Update the values of the variables in italics according to the deployment needs:
++
+[source,bash,subs="attributes+,macros+,+quotes"]
+----
+batch
+
+set DB_DRIVER_NAME=_mariadb_
+set DB_USER=_username_
+set DB_PASSWORD=_password_
+set DB_DRIVER=_org.mariadb.jdbc.Driver_
+set DB_XA_DRIVER=_org.mariadb.jdbc.MariaDbDataSource_
+set DB_JDBC_URL=_jdbc:mariadb://jdbc-host/keycloak_
+set DB_EAP_MODULE=_org.mariadb_
+
+set FILE=/opt/eap/extensions/jdbc-driver.jar
+
+module add --name=$DB_EAP_MODULE --resources=$FILE --dependencies=javax.api,javax.resource.api
+/subsystem=datasources/jdbc-driver=$DB_DRIVER_NAME:add(
+  driver-name=$DB_DRIVER_NAME,
+  driver-module-name=$DB_EAP_MODULE,
+  driver-class-name=$DB_DRIVER,
+  driver-xa-datasource-class-name=$DB_XA_DRIVER
+)
+/subsystem=datasources/data-source=KeycloakDS:remove()
+/subsystem=datasources/data-source=KeycloakDS:add(
+  jndi-name=java:jboss/datasources/KeycloakDS,
+  enabled=true,
+  use-java-context=true,
+  connection-url=$DB_JDBC_URL,
+  driver-name=$DB_DRIVER_NAME,
+  user-name=$DB_USER,
+  password=$DB_PASSWORD
+)
+
+run-batch
+----
+. In this directory, build your image by typing the following command, replacing the `project/name:tag` with arbitrary name. `docker` can be used instead of `podman`.
++
+[source,bash,subs="attributes+,macros+,+quotes"]
+----
+podman build -t docker-registry-default/project/name:tag .
+----
+. After the build finishes, push your image to the registry used by OpenShift to deploy your image:
++
```

Comment:
I would drop the push command example below, and use link [to the official guide](https://docs.openshift.com/container-platform/4.3/registry/accessing-the-registry.html#registry-accessing-directly_accessing-the-registry) instead.

The thing is the procedure is a bit more complex than just plain push (from within the cluster add the `registry-viewer` & `registry-editor` roles, login to the registry, then push. From outside the cluster first expose the registry, then login etc.) 

IOW I would rather recommend to point to the official doc on how to push the image to the registry, than try to describe this.

Reply:
Good point, thanks for the link!